formats/json-jackson/src/main/java/io/cloudevents/jackson/CloudEventDeserializer.java

```diff
@@ -126,7 +126,13 @@ public JsonMessage(JsonParser p, ObjectNode node) {
 
                 // Now let's process the extensions
                 node.fields().forEachRemaining(entry -> {
-                    String extensionName = entry.getKey();
+                    String extensionName = entry.getKey().toLowerCase();
+
+                    // ignore not valid extension name
+                    if (!this.isValidExtensionName(extensionName)) {
+                        return;
+                    }
```

Comment:
This might be surprising or unwanted.

Can we make it an opt-in behavior?

Reply:
@pierDipi 
Thank you for your feedback
I added two options for jackson CloudEventDeserializer
https://github.com/cloudevents/sdk-java/pull/429/commits/cc4963e0cd6a8e83a30529cf2a9ceecc41f93e48